tests/integration/test_model.py

```diff
@@ -44,153 +35,165 @@ def check_trained_model(c, model_name=None):
     assert len(result_df["target"]) > 0
 
 
-@pytest.fixture()
-def training_df(c):
-    df = timeseries(freq="1d").reset_index(drop=True)
-    c.create_table("timeseries", df, persist=True)
-
-    return None
-
-
-@pytest.fixture()
-def gpu_training_df(c):
-    if dask_cudf:
-        df = timeseries(freq="1d").reset_index(drop=True)
-        df = dask_cudf.from_dask_dataframe(df)
-        c.create_table("timeseries", input_table=df)
-    return None
-
-
 # TODO - many ML tests fail on clusters without sklearn - can we avoid this?
 @xfail_if_external_scheduler
-def test_training_and_prediction(c, training_df):
-    c.sql(
+@pytest.mark.parametrize(
+    "gpu_client", [False, pytest.param(True, marks=pytest.mark.gpu)], indirect=True
+)
+def test_training_and_prediction(c, gpu_client):
+    gpu = "CUDA" in str(gpu_client.cluster)
+    timeseries = "gpu_timeseries" if gpu else "timeseries"
+
+    # cuML does not have a GradientBoostingClassifier
+    if not gpu:
+        c.sql(
+            """
+            CREATE MODEL my_model WITH (
+                model_class = 'GradientBoostingClassifier',
+                wrap_predict = True,
+                target_column = 'target'
+            ) AS (
+                SELECT x, y, x*y > 0 AS target
+                FROM timeseries
+                LIMIT 100
+            )
         """
-        CREATE MODEL my_model WITH (
-            model_class = 'sklearn.ensemble.GradientBoostingClassifier',
+        )
+        check_trained_model(c)
+
+    c.sql(
+        f"""
+        CREATE OR REPLACE MODEL my_model WITH (
+            model_class = 'LogisticRegression',
             wrap_predict = True,
+            wrap_fit = False,
             target_column = 'target'
         ) AS (
             SELECT x, y, x*y > 0 AS target
-            FROM timeseries
-            LIMIT 100
+            FROM {timeseries}
         )
     """
     )
+    check_trained_model(c, df_name=timeseries)
 
-    check_trained_model(c)
+    # TODO: In this query, we are using cuml.dask.linear_model.LinearRegression
+    # instead of cuml.linear_model.LinearRegression.
+    # Is there any way to assert that we are using the cuML Dask estimator
+    # (and not just the cuML estimator)?
```

Comment:
Is this a concern for just this specific model class, or throughout the test suite? In either case, as long as `wrap_predict=False` then something like

```python
assert "dask" in str(c.schema["root"].models["my_model"][0].__class__).lower()
```

Should work here

Reply:
Cool, thanks! The reason I wanted to have a check is because originally this code was part of `test_dask_cuml_training_and_prediction` which was specifically meant to test `cuml.dask.linear_model.LinearRegression` functionality.

With this PR, `LinearRegression` and [some others](https://github.com/dask-contrib/dask-sql/blob/9425286a326051613403fd03bc513ef302876a11/dask_sql/physical/utils/ml_classes.py#L32) map to the `cuml.dask` version automatically. I figured it was nicer to map to the Dask version of a model if it's available - what do you think?

Reply:
> I figured it was nicer to map to the Dask version of a model if it's available

That makes sense to me, this shouldn't be too limiting since users can always just specify the non-Dask version explicitly if they need to use it for some reason.

IMO unless there's some implicit fallback behavior in the `cuml.dask` classes when certain conditions aren't met such that they default to their non-Dask variants, I would find the checks a little superfluous, as the mapping to Dask classes and their import is relatively straightforward and I wouldn't expect it to change unless we specifically modified the mapping; if there's a case where this could change unexpectedly though would be happy to know more about it 🙂 and would recommend adding this check conditionally to `check_trained_model` so that it can easily be extended to other cases where we might want to assert the Dask class is being used.